hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -2741,7 +2741,8 @@ private void removeCompactedfiles(Collection<HStoreFile> compactedfiles)
             LOG.trace("Closing and archiving the file {}", file);
             // Copy the file size before closing the reader
             final long length = r.length();
-            r.close(true);
+            r.close(file.getCacheConf() != null ?
```

Comment:
This area will be getting called when we close a store as well as when the CompactionDischarger finds unreferenced compacted files..  CompactionDischarger  is working in an async way anyways.  But ya as per ur explanation, during close() we wanted to keep it this was as in patch.

Reply:
Sorry I was not clear in above comment. My ask was this. 
This is called from closeAndArchiveCompactedFiles() and HStore#close(). 
The former is called in an async way by the Discharger thread anyways.  When this area is called by that flow, we can preserve what is there today (Passing true).  Close time ya lets honor the config.  Sounds ok?

Ideally speaking we should have handled whole these in another way. Within the Cache implementation, the evict by hfile would have been an async impl. So the caller wont get blocked.  But ya lets handle that in another jira.

Reply:
Thank you for the clarification. I understood that. Will change the patch according to your review. Thanks.